scripts/src/admin/translations.ts

```diff
@@ -0,0 +1,162 @@
+	const path = `${contentType}:${key.replace(KEY_TO_PATH_REGEX, "$1")}`;
+	if (EXCLUDED.includes(path) || EXCLUDED.includes(contentType)) {
+		return;
+	}
+	return {
+		Type: contentType,
+		Key: key,
+		Default: str,
+		Translation: "",
+		"Character Limit": CHARACTER_LIMITS[path],
+	};
+}
+
+function constructRowsFromArray(keyBase: string, arr: any[], rowConstructor: (key: string, str: string) => CsvRow) {
+	let result: CsvRow[] = [];
+	arr.forEach((item: any, index: number) => {
+		const key = `${keyBase}[${index}]`;
+		if (typeof item === "string" && item !== "") {
+			result.push(rowConstructor(key, item));
+			return;
+		}
+		if (Array.isArray(item)) {
+			result = result.concat(constructRowsFromArray(key, item, rowConstructor));
+			return;
+		}
+
+		if (typeof item === "object") {
+			result = result.concat(constructRowsFromObj(key, item, rowConstructor));
+			return;
+		}
+	});
+	return result;
+}
+
+function constructRowsFromObj(keyBase: string, obj: { [key: string]: any }, rowConstructor: (key: string, str: string) => CsvRow) {
+	let result: CsvRow[] = [];
+	Object.keys(obj).forEach((itemKey: any) => {
+		const item = obj[itemKey];
+		const key = `${keyBase}.${itemKey}`;
+		if (typeof item === "string" && item !== "") {
+			result.push(rowConstructor(key, item));
+			return;
+		}
+		if (Array.isArray(item)) {
+			result = result.concat(constructRowsFromArray(key, item, rowConstructor));
+			return;
+		}
+
+		if (typeof item === "object") {
+			result = result.concat(constructRowsFromObj(key, item, rowConstructor));
+			return;
+		}
+	});
+	return result;
+}
+
+async function getCsvRowData() {
+	const allOffers = await Offer.find();
+	const allContent = await OfferContent.find({ select: ["offerId", "content", "contentType"] } as FindManyOptions<OfferContent>);
+	let rows: CsvRow[] = [];
+	allOffers.forEach(offer => {
+		if (offer.type === "spend") {
+			return;
+		}
+		const offerId = offer.id;
+		const offerContent: OfferContent = allContent.filter(obj => obj.offerId === offerId)[0];
+		// quote unquoted template values
+		const escapedOfferContent = offerContent.content.replace(/:\s(\${[\w\.-_]+})/g, ": \"$1\"");
+		const offerContentContent: OfferContentContent = JSON.parse(escapedOfferContent);
+		const boundConstructRow = constructRow.bind({}, offerContent.contentType);
+		let keyBase = `offer:${offerId}`;
+		rows = rows.concat([
+			boundConstructRow(`${keyBase}:title`, offer.meta.title),
+			boundConstructRow(`${keyBase}:description`, offer.meta.description),
+			boundConstructRow(`${keyBase}:orderTitle`, offer.meta.order_meta.title),
+			boundConstructRow(`${keyBase}:orderDescription`, offer.meta.order_meta.description),
+		]);
+		keyBase = `offer_contents:${offerId}`;
+		if (offerContentContent.pages) {
+			rows = rows.concat(constructRowsFromArray(`${keyBase}:content:pages`, offerContentContent.pages, boundConstructRow));
+		}
+		if (offerContentContent.confirmation) {
```

Comment:
Isn't this for spend?

Reply:
Since I've already written this, I'm leaving the spend support.